__tests__/app.test.js

```diff
@@ -156,3 +156,75 @@ describe('GET /api/articles/:article_id/comments', () =>{
     }) 
 })
 
+describe('POST /api/articles/:article_id/comments', () => {
+    test('201: user can post new comments under specific articles', () => {
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this comment got posted'
+        }
+        return request(app).post('/api/articles/3/comments').send(newComment).expect(201).then(({ body }) => {
+            const { comment } = body
+            expect(comment).toMatchObject({
+                comment_id: 19,
+                author: 'butter_bridge',
+                body: 'this comment got posted',
+                votes: 0,
+                article_id: 3,
+                created_at: expect.any(String)
+            })
+        })
+    })
+    test('400: returns error when user inputs non-existent ID', () => {
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this comment did not get posted'
+        }
+        return request(app).post('/api/articles/999/comments').send(newComment).expect(400).then(({ body }) => {
+            const { msg } = body
+            expect(msg).toBe('Bad Request')
+        })
+    })
```

Comment:
Almost perfect - apart from the point that this should be a 'not found'. A bad request is more like the wrong data type - this is a fine request but there's nothing found for id 999